### ISSUE TITLE
Hide the RetentionManager in the API?

### ISSUE THREAD

**Issue by psh** [ChuckerTeam/chucker#31]
I believe that it would be possible to hide the `RetentionManager` from the public API and treat it as an implementation detail, so users would only need to supply a _retention period_ and internally the `ChuckCollector` manages an instance of it.
```kotlin
collector = ChuckCollector(this)
        .showNotification(true)
        .retentionManager(RetentionManager(this, ChuckCollector.Period.ONE_HOUR))
```
becoming 
```kotlin
collector = ChuckCollector(this)
        .showNotification(true)
        .retentionPeriod(ChuckCollector.Period.ONE_HOUR)
```
I am assuming that you would always want to use the same `context` for both the collector and the retention manager.  Can you imagine any circumstance where this wouldn't be the case and the exposed constructor would be useful?

Do you feel this would be a good way to evolve the API?

**Do you want to develop this feature yourself?**
Yes, I'd be happy to.

**Comment by cortinico** [comment]
Hey @psh 

This shoulds like a good idea to me :)
Please note that we also recently migrated all the retention mechanism to Room, so it should be easier to integrate with your proposal. Would be happy/free to code this feature?

**Comment by cortinico** [comment]
Closed by #36
